src/server/views/dashboard/lists-edit-annual-review-date.njk

```diff
@@ -0,0 +1,68 @@
+{% from "button/macro.njk" import govukButton %}
+{% extends "./dashboard.njk" %}
+
+{% block aside %}
+{% endblock %}
+
+{% block dashboard %}
+
+  {% if error.text %}
+    {{ dashboardMacros.errorBanner(error) }}
+  {% endif %}
+
+  <form class="form" action="{{ dashboardRoutes.listsEditAnnualReviewDate.replace(':listId', list.id) }}" method="post" required>
+    <input type="hidden" name="_csrf" id="_csrf" value="{{ csrfToken }}" />
+
+    <h2 class="govuk-heading-m" style="color: #b1b4b6;">{{list.country.name}}</h2>
+
+    <h1 class="govuk-heading-l">
+      Change annual review start date
+    </h1>
+
+    <p class="govuk-body">The next funeral directors list annual review starts <strong>{{ annualReviewStartDate }}.</strong></p>
+
+    <div class="govuk-form-group {{ "govuk-form-group--error" if error.text }}" id="annualReveiwDateForm">
+        <h2 class="govuk-heading-m govuk-!-margin-0">Enter a new start date</h2>
+        <fieldset class="govuk-fieldset" role="group" aria-describedby="annual-review-date">
+          <legend class="govuk-fieldset__legend govuk-fieldset__legend--l">
+          </legend>
+          <div id="annual-review-date" class="govuk-hint">
+              The new date must be within 6 months of today’s date
+          </div>
+          {% if error.text %}
+            <span class="govuk-error-message">
+              <span class="govuk-visually-hidden">Error:</span> {{ error.text }}
+            </span>
+          {% endif %}
+          <div class="govuk-date-input" id="annual-review-date">
+            <div class="govuk-date-input__item">
+              <div class="govuk-form-group">
+                <label class="govuk-label govuk-date-input__label" for="annual-review-date-day">
+                    Day
+                </label>
+                <input class="govuk-input govuk-date-input__input govuk-input--width-2 {{ "govuk-input--error" if error.text }}" data-testid="day" id="annual-review-date-day" name="annual-review-date-day" type="text" inputmode="numeric">
+              </div>
+            </div>
+            <div class="govuk-date-input__item">
+              <div class="govuk-form-group">
+                <label class="govuk-label govuk-date-input__label" for="annual-review-date-month">
+                    Month
+                </label>
+                <input class="govuk-input govuk-date-input__input govuk-input--width-2 {{ "govuk-input--error" if error.text }}" data-testid="month" id="annual-review-date-month" name="annual-review-date-month" type="text" inputmode="numeric">
+              </div>
+            </div>
+          </div>
+        </fieldset>
```

Comment:
So I tried to replace the date input with this:
```
{{ govukDateInput({
  id: "passport-issued",
  namePrefix: "passport-issued",
  fieldset: {
    legend: {
      text: "When was your passport issued?",
      isPageHeading: true,
      classes: "govuk-fieldset__legend--l"
    }
  },
  errorMessage: {
    text: error.text
  },
  items: [
    {
      classes: "govuk-input--width-2",
      name: "day",
      attributes: {
        id: "day",
        "data-testid": "day"
      }
    },
    {
      classes: "govuk-input--width-2",
      name: "month",
      attributes: {
        id: "month",
        "data-testid": "month"
      }
    }
  ]
}) }}
``` 

and some weird things happen.


1. There's always a red bar on the side.

<img width="637" alt="Screenshot 2022-10-27 at 21 17 57" src="https://user-images.githubusercontent.com/1377253/198389785-429d9214-4711-41fc-9d2d-966378827b18.png">

I think this is is because of
```
  errorMessage: {
    text: error.text
  },
  ```
  
  But would this mean I need to have some sort of condition for `errorMessage`?
  
  
  2. The form no longer submits 😢 
  
  
<img width="598" alt="Screenshot 2022-10-27 at 21 18 03" src="https://user-images.githubusercontent.com/1377253/198389955-536e58cf-3bad-4b74-a263-aaa0854edae9.png">

Even with matching attributes no matter what I put in the inputs I always get the above error.

I'm not sure what I'm doing wrong.

Reply:
I assume errorMessage needs to be falsy entirely. based on your snippet it looks truthy to me.

This is the documentation:

>Options for the error message component. The error message component will not display if you use a falsy value for errorMessage, for example false or null. See [errorMessage](https://design-system.service.gov.uk/components/error-message/#options-error-message-example).

errorMessage needs to be falsy. However doing 

```
  errorMessage: {
    text: error.text
  },
```

would mean that errorMessage is truthy, there is a defined object (even though text is undefined). 

e.g.
```
const errorMessage = { text: undefined } // is truthy 
const falsyError = undefined // falsy 

```

Reply:
Also, I noticed in this snippet

```
  id: "passport-issued",
  namePrefix: "passport-issued",
```

the rendered id for your day field is going to be something like `passport-issued-day`.